### ISSUE TITLE
Board does not display anything while server is updating data

### ISSUE THREAD

**Issue by jdbaptista** [zinkwazi/traffic-pcb-sb#42]
This is likely due to the fact that the board writes to the file directly while querying data instead of writing to a temporary file and copying it all at once to the data file. This actually breaks both north and south directions at the same time, which is interesting because only one file is updated at a time). Here are example logs from clicking the change direction button in v1_0_2 while the server is updating data:

```
I (30710) esp-x509-crt-bundle: Certificate validated
E (31350) dot_worker: Failed to retrieve segment speeds from server
I (39170) dot_worker: Clearing North...
I (39370) dot_worker: Refreshing North...
E (39380) dot_worker: Failed to retrieve segment speeds from server
I (41510) dot_worker: Clearing South...
I (41710) dot_worker: Refreshing South...
E (41710) transport_base: poll_write select error 104, errno = Connection reset by peer, fd = 54
W (41720) transport_base: Poll timeout or error, errno=Success, fd=54, timeout_ms=5000
E (41720) HTTP_CLIENT: Error write request
E (41740) TomTom: failed to open connection
E (41750) dot_worker: Failed to retrieve segment speeds from server
```

It may be that the server cannot handle running and handling a file access at the same time? Although this seems to be disproven by accessing bearanvil.com while running the script.

**Comment by jdbaptista** [comment]
The break is due to esp_http_client_cleanup() not being called after the server responds with a non-200 code. This was fixed as of 7221253.

**Comment by jdbaptista** [comment]
I'm not exactly sure what the issue here was or why it was actually an issue, but as of 1a20074, if any request to the server fails, data stored in non-volatile storage from the most recent request is used and the error LED is flashed. I'm closing this unless it proves to be a bigger issue that affects functionality.
